### PR TITLE
feat(tools): create package plugins

### LINKED ISSUE #29: feat(tools): create package plugins
### Passive Mode:
`/tools/finders/plugins/passive.go`

To complete this card, in passive mode, you must obtain the information from the plugins found in the index (main page) of the site.

Example:
Image

Note that at the end of the URL's there is a `?ver=3.4.35`, this is an indication of the version of the plugin that is installed on the site; It may happen that there are more HTML tags in the same format with different versions, in which case you must also return to the user a confidentiality status of the information, which must be increased with each compatible version found.
There are also cases where there is no `?ver=` at the end of the URL, and the information must still be returned.

Tip: retrieve the plugins and version using the url as part of the regular expression.

### Aggressive Mode:
`tools/finders/plugins/aggressive.go`

Here in aggressive mode, information will also be collected from the index of the `wp-content/plugins` directory, which is where all Wordpress plugins are listed; Therefore, we can list all the plugins on the site and if we are lucky, we may also be able to access the `index of` directory of these plugins.
Image

Within the directory of a plugin we can have files such as `readme.txt`, `changelog.txt` etc., which will list the current version of the plugin; The versions obtained within these files must have greater confidentiality than those found on the index (main page).

Image

Note: Due to a lack of competent administration on sites that use Wordpress they allow access to this `index-of`.

It is interesting that the information obtained is returned in this format or in an idea similar to this:

```
- Plugin name
  | Plugin name confidentiality
  | Plugin name match
  | Version 3.1.2:
    | Version confidentiality
    | Version matches
    | Vulnerabilities found for version 3.1.2
  | Version 2.5.4:
    | Version confidentiality
    | Version matches
    | Vulnerabilities found for version 2.5.4
  | Version 2.0.1:
    | Version confidentiality
    | Version matches
    | Vulnerabilities found for version 2.0.1
```

Example:

```
- jetpack
  | Confidence: 100%
  | Match:
    | - https://example.com/wp-content/plugins/jetpack/_inc/genericons/genericons/genericons.css?ver=3.1
    | - https://example.com/wp-content/plugins/
    | - https://example.com/wp-content/plugins/reader/changelog.txt
  | Version: 3.1:
    | Confidence: 10%
    | Match:
      | - https://example.com/wp-content/plugins/jetpack/_inc/genericons/genericons/genericons.css?ver=3.1 (?ver=3.1)
  | Version 12.9.1:
    | Confidence: 100%
    | Match:
      | - https://example.com/wp-content/plugins/jetpack/CHANGELOG.md (## 12.9.1 - 2023-12-18)
      | - https://example.com/wp-content/plugins/jetpack/readme.txt (Stable tag: 12.9.1)
```

### PR DESCRIPTION
**Pull request by hax1zi**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

Issue Number: 10

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the new behavior?
system of searching plugins version

### CODE REVIEW

**Review by ffx64 — DISMISSED** [review]
LGTM

File: tools/finders/plugins/agressive.go
Comment (rafael-dev2021) [inline tools/finders/plugins/agressive.go:1]: These code snippets are repeated 5 times in the same class,
response := ghttp.GET(url)
url += "/wp-content/plugins/"

Wouldn't it be better to encapsulate them in two functions and concatenate them with the suffixes "plugin", "/readme.txt" and "/changelog.txt"?

**Review by ffx64 — APPROVED** [review]
LGTM

**Review by rafael-dev2021 — APPROVED** [review]
LGTM

**Pull request merged**